sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobItem.java

```diff
@@ -48,6 +49,25 @@ public BlobItem(BlobItemInternal blobItemInternal) {
                 this.tags.put(tag.getKey(), tag.getValue());
             }
         }
+        this.objectReplicationSourcePolicies = null;
+        if (blobItemInternal.getObjectReplicationMetadata() != null) {
+            this.objectReplicationSourcePolicies = new HashMap<>();
+            for (Map.Entry<String, String> p : blobItemInternal.getObjectReplicationMetadata().entrySet()) {
+                String orString = p.getKey();
+                String str = orString.startsWith("or-") ? orString.substring(3) : orString;
+                String[] split = str.split("_");
+                String policyId = split[0];
+                String ruleId = split[1];
+                if (objectReplicationSourcePolicies.containsKey(policyId)) {
+                    objectReplicationSourcePolicies.get(policyId)
+                        .putRuleAndStatus(ruleId, p.getValue());
+                } else {
+                    ObjectReplicationPolicy policy = new ObjectReplicationPolicy(policyId);
+                    policy.putRuleAndStatus(ruleId, p.getValue());
+                    objectReplicationSourcePolicies.put(policyId, policy);
+                }
+            }
+        }
```

Comment:
This needs to find new home. We need to decouple model from "implementation model" per APIView feedback.
Anyway, it's a good practice to NOT place such logic inside constructors. BlobItem should ideally be plain data structure and this logic should move to some "transformer/mapper".